### PR TITLE
Re-structure running the workflow docs

### LINKED ISSUE #319: Restructure running the workflow section in README
**Please provide some background on the proposed additions or changes.**
There two issues but they are related to one another and we should consider both the issues for the solution. 

During evals:
1) when users were ready to run the workflow, they consistently modified the paths in both the config file as well as the command. It is sufficient to only modify it one place, but that is not coming through with how the information is presented. 

2) Modifying the paths on the command line is a pain. Unless someone is a super-user, people are more likely to copy the command and then modify it with the parameters. 
 

**Please describe the proposed solution.**

1) We restructure the documentation so it is `configure config file` -> `run pipeline`. We encourage people to modify the paths (or other parameters) in the config file. 

2) The default running the workflow command omits the path options.

3) We can tuck away command line options in a different file. We can present the options similar to a [man {cmd}](https://man7.org/linux/man-pages/man1/ls.1.html) document.

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #319 and #320

**What is the purpose of these changes?**

This PR re-structures the "running the workflow" section of the core `README.md` file and adds a "configure config file" step just before.

**What changes did you make?**

In this PR, modifications are made to the core `README.md` file to encourage users to manually change parameters in the config file using a text editor.
The `--config` flag is also removed from the running the workflow example since we added a `4. Configure Config File` section.

The option to modify the parameters at the command line is instead noted in a separate `additional-docs/command-line-options.md` doc where an example on doing this is provided and the parameters that can be modified are listed.

The sections `2. Input Data Format` and `3. Metadata File Format` were also renamed to `2. Verify Input Data Format` and `3. Create Metadata File`, respectively.

**Any comments, concerns, or questions important for reviewers**
- What do you think about the added documentation? It's not exactly a man {cmd} document as mentioned in https://github.com/AlexsLemonade/scpca-downstream-analyses/issues/319#issue-1622114684 (wanted to keep things consistent across the repo and provide the example we originally had in the main README), so please let me know if you have suggestions for refactoring here!

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,20 +60,14 @@ To run the core analysis workflow you will want to implement the following steps
...
-1. [Create a metadata tab-separated value (TSV) file](#3-metadata-file-format) that defines the sample id, library id, and filepath associated with the pre-processed `SingleCellExperiment` files to be used as input for the workflow.
-2. Open terminal to run the workflow using the following snakemake command and the `--config` flag to adjust the `results_dir` and `project_metadata` parameters to point to your desired results directory and project metadata file that you created in step 3:
+4. [Create a metadata tab-separated value (TSV) file](#3-metadata-file-format) that defines the sample id, library id, and filepath associated with the pre-processed `SingleCellExperiment` files to be used as input for the workflow.
+5. Configure the config file to adjust the `results_dir` and `project_metadata` parameters to point to your desired results directory and project metadata file that you created in step 4.
```
Comment (allyhawkins) [inline README.md:64]:
```suggestion
5. Configure the config file to adjust the `results_dir` and `project_metadata` parameters to point to the full path to your desired results directory and project metadata file that you created in step 4.
```
Let's just make sure we specify the full path throughout.

File: README.md [hunk 2]
```diff
@@ -13,14 +13,15 @@
     - [How to install Snakemake](#how-to-install-snakemake)
   - [c) Additional dependencies](#c-additional-dependencies)
     - [Snakemake/conda installation](#snakemakeconda-installation)
-- [2. Input data format](#2-input-data-format)
-- [3. Metadata file format](#3-metadata-file-format)
-- [4. Running the workflow](#4-running-the-workflow)
+- [2. Verify Input Data Format](#2-verify-input-data-format)
+- [3. Create Metadata File](#3-create-metadata-file)
+- [4. Configure Config File](#4-configure-config-file)
```
Comment (allyhawkins) [inline README.md:18]: Just noting that the rest of the headers are in sentence case except for these new changes, so let's change them back to sentence case.

File: README.md [hunk 3]
```diff
@@ -299,7 +263,23 @@ snakemake --cores 2 \
 Also note that new changes should be merged through a pull request to the `development` branch.
 Changes will be pushed to the `main` branch once changes are ready for a new release (per the [release checklist document](.github/ISSUE_TEMPLATE/release-checklist.md)).
 
-## 5. Expected output
+## 5. Running the workflow
+
+You can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example:
```
Comment (allyhawkins) [inline README.md:268]: I would add a sentence here that you can run the workflow after you have successfully modified the required parameters in the config file.

File: additional-docs/command-line-options.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+| `prob_compromised_cutoff` | the maximum probability of a cell being compromised as calculated by [miQC](https://bioconductor.org/packages/release/bioc/html/miQC.html), which is required when the `filtering_method` is set to `miQC` in the project metadata | 0.75 |
+| `filter_genes` | a binary value indicating whether or not to perform gene fitering | `FALSE` |
+| `gene_detected_row_cutoff` | the percent of cells a gene must be detected in; genes detected are only filtered if `filter_genes` is set to `TRUE` | 5 |
+| `gene_means_cutoff` | mean gene expression minimum threshold; mean gene expression is only filtered if `filter_genes` is set to `TRUE` | 0.1 |
+| `mito_percent_cutoff` | maximum percent mitochondrial reads per cell threshold, which is only required when `filtering_method` is set to `manual` | 20 |
+| `min_gene_cutoff` | minimum number of genes detected per cell | 200 |
+| `umi_count_cutoff` | minimum unique molecular identifiers (UMI) per cell, which is only required when `filtering_method` is set to `manual` | 500 |
+
+
+## List of additional analysis module parameters that can be modified
```
Comment (allyhawkins) [inline additional-docs/command-line-options.md:53]: Same thing for these here, I don't think you need the whole list, you just want to show an example of running the workflow with the modification at the command line and then link back to the readme with the list of parameter options.

**Review by allyhawkins — COMMENTED** [review]
I like how these changes are coming together! I have a few suggestions about the command line section that I think will help clean it up, but also can you make the same edits to the GOI and the clustering section for running the workflow? All the instructions should mirror each other.

File: README.md [hunk 5]
```diff
@@ -188,17 +183,25 @@ Each library ID should have a unique `filepath`.
 |[View Example Metadata File](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/example-data/project-metadata/example-library-metadata.tsv)|
 |---|
 
-## 4. Running the workflow
+## 4. Configure config file
 
-We have provided an example [snakemake configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), [`config/config.yaml`](config/config.yaml) which sets the defaults for all parameters needed to run the workflow.
+We have provided an example [snakemake configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), [`config/config.yaml`](config/config.yaml) which defines all parameters needed to run the workflow.
```
Comment (dvenprasad) [inline README.md:188]: The way the sentence is phrased make me think that the link on snakemake configuration will link me to the example file. That combined with the low affordance on links in code blocks makes it difficult to get to the example config file. 

 ```suggestion
We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html)
```

**Review by dvenprasad — CHANGES_REQUESTED** [review]
The restructure is looking good! Definitely more streamlined. 
 Had a couple comments about linking
The big comment I had is about cutting extraneous information in `4. Configure the config file`, so it is easy to parse and what users need to do is front and center.

File: optional-clustering-analysis/README.md [hunk 6]
```diff
@@ -76,10 +56,15 @@ For example, if you would like a range of `5:25` to be tested in increments of 5
 
 If you are working with data from the ScPCA portal, see our guide on preparing that data to run the clustering workflow [here](../additional-docs/working-with-scpca-portal-data.md).
 
-## Parameters and config file
+## Configure config file
 
 As in the main core workflow, we have provided a [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config/config.yaml` which sets the defaults for project-specific parameters needed to run the clustering workflow.
 
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
```
Comment (dvenprasad) [inline optional-clustering-analysis/README.md:63]: similar comment to https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/328#discussion_r1154667947

 Restructure to make it more instructional and tuck away the parameters in a different file.

File: optional-clustering-analysis/README.md [hunk 7]
```diff
@@ -76,10 +56,15 @@ For example, if you would like a range of `5:25` to be tested in increments of 5
 
 If you are working with data from the ScPCA portal, see our guide on preparing that data to run the clustering workflow [here](../additional-docs/working-with-scpca-portal-data.md).
 
-## Parameters and config file
+## Configure config file
 
 As in the main core workflow, we have provided a [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config/config.yaml` which sets the defaults for project-specific parameters needed to run the clustering workflow.
```
Comment (dvenprasad) [inline optional-clustering-analysis/README.md:61]: The way sentence is phrased makes me think that `configuration file` links to the config file and not the snakemake docs. 

Would recommend a similar change as this https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/328#discussion_r1154675406

File: optional-clustering-analysis/README.md [hunk 8]
```diff
@@ -129,6 +98,23 @@ The parameters found in the `config/cluster_config.yaml` file can be optionally
 |[View Clustering Config File](../config/cluster_config.yaml)|
 |---|
 
+## Running the workflow
+
+The execution file with the clustering Snakemake workflow is named `cluster.snakefile` and can be found in the root directory. To tell snakemake to run the specific clustering workflow be sure to use the `--snakefile` or `-s` option followed by the name of the snakefile, `cluster.snakefile`. 
+After you have successfully modified the required parameters in the config file and navigated to within the root directory of the `scpca-downstream-analyses` repository, you can run the clustering Snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example:
```
Comment (dvenprasad) [inline optional-clustering-analysis/README.md:104]:
```suggestion

After you have successfully modified the required parameters in the config file and navigated to within the root directory of the `scpca-downstream-analyses` repository, you can run the clustering Snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 
```

For better chunking

File: optional-goi-analysis/README.md [hunk 9]
```diff
@@ -72,10 +50,15 @@ This file should contain at least one column named `gene_id` with the relevant g
 
 If working with data from the ScPCA portal, see our guide on preparing that data to run the genes of interest workflow [here](./additional-docs/working-with-scpca-portal-data.md).
 
-## Parameters and config file
+## Configure config file
 
 As in the main core workflow, we have provided a [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config/config.yaml` which sets the defaults for project-specific parameters needed to run the genes of interest workflow.
```
Comment (dvenprasad) [inline optional-goi-analysis/README.md:55]: same comment as https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/328#discussion_r1154675406

File: optional-goi-analysis/README.md [hunk 10]
```diff
@@ -72,10 +50,15 @@ This file should contain at least one column named `gene_id` with the relevant g
...
 
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
```
Comment (dvenprasad) [inline optional-goi-analysis/README.md:57]: Similar comment to https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/328#discussion_r1154667947

Restructure to make it more instructional and tuck away the parameters in a different file.

File: optional-goi-analysis/README.md [hunk 11]
```diff
@@ -162,6 +104,24 @@ The following gene mapping parameters found in the `config/goi_config.yaml` file
 |[View Genes of Interest Config File](../config/goi_config.yaml)|
 |---|
 
+## Running the workflow
+
+The execution file with the genes of interest Snakemake workflow is named `goi.snakefile` and can be found in the root directory.
+To tell snakemake to run the specific genes of interest workflow be sure to use the `--snakefile` or `-s` option followed by the name of the snakefile, `goi.snakefile`.
+After you have successfully modified the required parameters in the config file and navigated to within the root directory of the `scpca-downstream-analyses` repository, you can run the clustering Snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example:
```
Comment (dvenprasad) [inline optional-goi-analysis/README.md:111]:
```suggestion

After you have successfully modified the required parameters in the config file and navigated to within the root directory of the `scpca-downstream-analyses` repository, you can run the clustering Snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 
```
Chunking!

**Review by dvenprasad — CHANGES_REQUESTED** [review]
Oops didn't see the new files come in. I had similar comments as for these as the README, which is to make the config section more instructional and to tuck away the additional parameter instructions.

**Comment by cbethell** [comment]
Thanks @dvenprasad and @allyhawkins for the reviews!! I've implemented all of your suggestions and also added a separate `additional-parameters.md` doc in the last set of updates. I think the structure looks much better now, let me know what you think!

File: README.md [hunk 12]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
-### Project-specific parameters
+We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
+Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
 
-There are a set of parameters included in the `config/config.yaml` file that will need to be specified when running the workflow.
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
+There are a set of parameters included in the `config/config.yaml` file that will **need to be specified** when running the workflow with your own data.
 These parameters are specific to the project or dataset being processed.
-These include the following parameters:
+These project-specific parameters can be found under the [`Project-specific parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Processing parameters` section](./config/config.yaml#L11).
```
Comment (allyhawkins) [inline README.md:190]:
```suggestion
These required parameters can be found under the [`Required parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Additional parameters` section](./config/config.yaml#L11).
```
Now that we have broken out the additional parameters, should we rename this to be required vs. additional? To me that is more straight forward and might be less confusing to someone.

File: README.md [hunk 13]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
+Therefore, if you would like to test this workflow using the example data, you can continue to the next step, running the workflow, without modifying the config file.
 
-```
-snakemake --cores 2 \
-  --use-conda \
-  --config results_dir="<RELATIVE PATH TO RESULTS DIRECTORY>" \
-  project_metadata="<RELATIVE_PATH TO YOUR PROJECT METADATA TSV>" \
-  mito_file="<FULL PATH TO MITOCHONDRIAL GENES TXT FILE>"
-```
+The config file also contains processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc.
```
Comment (allyhawkins) [inline README.md:204]:
```suggestion
The config file also contains additional processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc. 
```

File: README.md [hunk 14]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
-```
-snakemake --cores 2 \
-  --use-conda \
-  --config results_dir="<RELATIVE PATH TO RESULTS DIRECTORY>" \
-  project_metadata="<RELATIVE_PATH TO YOUR PROJECT METADATA TSV>" \
-  mito_file="<FULL PATH TO MITOCHONDRIAL GENES TXT FILE>"
-```
+The config file also contains processing parameters like cutoffs for minimum genes detected, minimum unique molecular identifiers (UMI) per cell, etc. 
+We have set default values for these parameters. 
+Learn more about the [processing parameters](./additional-docs/additional-parameters.md#core-analysis-parameters) and how to modify them.
```
Comment (allyhawkins) [inline README.md:206]:
```suggestion
Learn more about the [additional processing parameters](./additional-docs/additional-parameters.md#core-analysis-parameters) and how to modify them.
```

File: additional-docs/additional-parameters.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
+# Additional Workflow Parameters
+
+Below are additional parameters that are needed to run the analysis workflows in this repository.
```
Comment (allyhawkins) [inline additional-docs/additional-parameters.md:3]:
```suggestion
Below is a list of all additional parameters that are needed to run the analysis workflows in this repository.
These parameters are all included in the config files and can optionally be altered when running the workflows, but are not required to be changed.
```
I don't know how I feel about this wording, but you want to say something about how these are all additional parameters and that you shouldn't need to change any of these parameters, but users can if they would like.

File: additional-docs/command-line-options.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Running the core workflow at the command line
+
+In [step 4](../README.md#4-configure-config-file) of the main `README.md` file of this repository, we note that there are parameters that need to be modified in the provided config file to successfully run the workflow.
+There we recommend modifying the `config/config.yaml` file manually.
+
+All of the parameters in the config file can also be modified at the command line by using the [`--config` flag](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
+The below code is an example of running the Snakemake workflow using the project-specific parameters.
```
Comment (allyhawkins) [inline additional-docs/command-line-options.md:21]:
```suggestion
The below code is an example of running the Snakemake workflow using the required project-specific parameters.
```

File: additional-docs/command-line-options.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+snakemake --cores 2 \
+  --use-conda \
+  --config results_dir="<FULL PATH TO RESULTS DIRECTORY>" \
+  project_metadata="<FULL PATH TO YOUR PROJECT METADATA TSV>"
+```
+
+You will want to replace the paths for both `results_dir` and `project_metadata` to successfully run the workflow in the example above.
+Where `results_dir` is the full path to the results directory where all results from running the workflow will be stored and `project_metadata` is the full path to the TSV file containing the relevant information about your input files.
+
+You can find a list of parameters that can be modified at the command line for the core workflow under the [`project-specific parameters`](../README.md#project-specific-parameters) and [`processing parameters`](../README.md#processing-parameters) sections of the `README.md`.
```
Comment (allyhawkins) [inline additional-docs/command-line-options.md:33]: This isn't the right link anymore right? They should be directed to the additional parameters doc?

File: optional-clustering-analysis/README.md [hunk 18]
```diff
@@ -76,58 +54,48 @@ For example, if you would like a range of `5:25` to be tested in increments of 5
...
+Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
 
-### Project-specific parameters
-
-There are a set of parameters included in the `config/config.yaml` file that will need to be specified when running the workflow.
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
+There are a set of parameters included in the `config/config.yaml` file that will **need to be specified** when running the workflow with your own data.
 These parameters are specific to the project or dataset being processed.
-These include the following parameters:
+These project-specific parameters can be found under the [`Project-specific parameters` section](../config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found in [`cluster_config.yaml`](../config/cluster_config.yaml).
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:65]:
```suggestion
These required project-specific parameters can be found under the [`Project-specific parameters` section](../config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found in [`cluster_config.yaml`](../config/cluster_config.yaml).
```
I think we might want to change this config file to also be divided in required and additional?

**Review by allyhawkins — COMMENTED** [review]
This mostly looks good, my major comment left here is that I want to make sure we emphasize what is a required parameter vs. what is an additional optional parameter. I think rather than using the phrasing "project specific" and "processing", it might be helpful to use "required" and "optional".

File: README.md (hunk 12 above)
Comment (dvenprasad) [reply README.md:190]: I think we should continue to call them Project-specific and Processing parameters as it is more descriptive of what they are. 

Unless I am mistaken, Project specific parameters need to changed to match user's data while the processing parameters do not need to be. 

I have made suggestions as to how we can streamline this paragraph to do that.

File: README.md [hunk 19]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
+## 4. Configure config file
 
-### Project-specific parameters
+We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
+Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
 
-There are a set of parameters included in the `config/config.yaml` file that will need to be specified when running the workflow.
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
+There are a set of parameters included in the `config/config.yaml` file that will **need to be specified** when running the workflow with your own data.
 These parameters are specific to the project or dataset being processed.
```
Comment (dvenprasad) [inline README.md:189]:
```suggestion

```

File: README.md (hunk 12 above)
Comment (dvenprasad) [inline README.md:190]:
```suggestion
The config file contains two sets of parameters:

- **[Project-specific Parameters](./config/config.yaml#L3)** : This set of parameters are for specifying dataset or project related details. 
These parameters are **required** to run the workflow on your data.

- **[Processing Parameters](./config/config.yaml#L11)**: This set of parameters specify configurations for the TODO:say what it is for here
These parameters are set with default values.
You can change them to explore your data but it is optional.


You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
```

File: README.md [hunk 20]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
-
-See the [processing information documentation](./additional-docs/processing-information.md) for more information on the individual workflow steps and how the parameters are used in each of the steps.
+## 4. Configure config file
 
-### Project-specific parameters
+We have provided an [example snakemake configuration file](config/config.yaml), `config/config.yaml` which defines all parameters needed to run the workflow.
+Learn more about snakemake configuration files [here](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html).
 
-There are a set of parameters included in the `config/config.yaml` file that will need to be specified when running the workflow.
+You can modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
```
Comment (dvenprasad) [inline README.md:187]:
```suggestion

```

File: README.md (hunk 12 above)
Comment (dvenprasad) [reply README.md:190]: Does this preserve the meaning of the paragraph? Please add any bits of information that I missed. 

If this looks good, I would propagate this change to the clustering and GOI README too.

File: README.md [hunk 21]
```diff
@@ -188,118 +179,57 @@ Each library ID should have a unique `filepath`.
...
+There are a set of parameters included in the `config/config.yaml` file that will **need to be specified** when running the workflow with your own data.
```
[the remark this answers is not in the document]
Comment (allyhawkins) [reply README.md:188]: > I think we should continue to call them Project-specific and Processing parameters as it is more descriptive of what they are.

> Unless I am mistaken, Project specific parameters need to changed to match user's data while the processing parameters do not need to be.

I think this is fine to keep them this way, I would just want to be clear that the project specific parameters are the ones that are required to be changed while the processing ones are the ones that don't need to be changed. To me processing parameters would also mean they might be required to be changed, so I think as long as there is emphasis that those are optional then that's fine.

File: additional-docs/command-line-options.md (hunk 17 above)
Comment (cbethell) [reply additional-docs/command-line-options.md]: Good catch -- fixed these!

**Comment by cbethell** [comment]
Thanks again for the further feedback guys - I think this is coming along! I took most of your suggestions, restructured things the way Deepa suggested in https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/328#discussion_r1157718482 and propagated necessary changes to the module README files. I think this is ready for another round of review (🤞 I don't _think_ I missed anything that should have been addressed but please let me know if I did)!

File: README.md [hunk 22]
```diff
@@ -188,118 +179,60 @@ Each library ID should have a unique `filepath`.
...
 
-You can also modify the relevant parameters by manually updating the `config/config.yaml` file using a text editor of your choice.
-The project-specific parameters mentioned above can be found under the [`Project-specific parameters` section](./config/config.yaml#L3) of the config file, while the remaining parameters that can be optionally modified are found under the [`Processing parameters` section](./config/config.yaml#L11).
+## 5. Running the workflow
 
-We have also included example data in the `example-data` directory for testing purposes.
-The two example `_filtered.rds` files were both processed using the [`scpca-nf` workflow](https://github.com/AlexsLemonade/scpca-nf/blob/main/examples/README.md).
-The `config.yaml` file points to this example data by default.
-Therefore, if you would like to test this workflow using the example data, you can run snakemake with just the `--cores` and `--use-conda` flags as in the following example:
+After you have successfully modified the required parameters in the config file, you can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example:
```
Comment (allyhawkins) [inline README.md:218]:
```suggestion
After you have successfully modified the required project-specific parameters in the config file, you can run the snakemake workflow with just the `--cores` and `--use-conda` flags as in the following example: 
```
Sticking with Deepa's comment, we should keep the project/processing language but specify which ones are required to be modified. So I would make this slight modification for clarity.

File: additional-docs/command-line-options.md [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+- [List of core workflow parameters that can be modified](#list-of-core-workflow-parameters-that-can-be-modified)
+- [List of additional analysis module parameters that can be modified](#list-of-additional-analysis-module-parameters-that-can-be-modified)
+  - [Clustering analysis parameters](#clustering-analysis-parameters)
+  - [Genes of interest analysis parameters](#genes-of-interest-analysis-parameters)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Running the core workflow at the command line
+
+In [step 4](../README.md#4-configure-config-file) of the main `README.md` file of this repository, we note that there are parameters that need to be modified in the provided config file to successfully run the workflow.
```
Comment (allyhawkins) [inline additional-docs/command-line-options.md:17]:
```suggestion
In [step 4](../README.md#4-configure-config-file) of the main `README.md` file of this repository, we note that there are required project-specific parameters that need to be modified in the provided config file to successfully run the workflow.
```

**Review by allyhawkins — APPROVED** [review]
This looks so much better! I made some minor wording suggestions to keep the language consistent, but other than that you've addressed all the comments from my end.

**Review by dvenprasad — APPROVED** [review]
Just a some minor formatting and presentation comments. Otherwise all of comments I had are addressed! 
🐘 🐘 🐘

**Pull request merged**